modules/sharedaddy/sharing-sources.php

```diff
@@ -1200,31 +1216,35 @@ public function __construct( $id, array $settings ) {
 			}
 		}
 
-		if ( isset( $settings['url'] ) )
+		if ( isset( $settings['url'] ) ) {
 			$this->url = $settings['url'];
+		}
 	}
 
 	public function get_name() {
 		return $this->name;
 	}
 
 	public function get_display( $post ) {
-		$str = $this->get_link( $this->get_process_request_url( $post->ID ), esc_html( $this->name ), sprintf( __( 'Click to share on %s', 'jetpack' ), esc_attr( $this->name ) ), 'share='.$this->id );
+		$str = $this->get_link( $this->get_process_request_url( $post->ID ), esc_html( $this->name ), sprintf( __( 'Click to share on %s', 'jetpack' ), esc_attr( $this->name ) ), 'share=' . $this->id );
 		return str_replace( '<span>', '<span style="' . esc_attr( 'background-image:url("' . addcslashes( esc_url_raw( $this->icon ), '"' ) . '");' ) . '">', $str );
 	}
 
 	public function process_request( $post, array $post_data ) {
 		$url = str_replace( '&amp;', '&', $this->url );
+		$url = str_replace( '%post_id%', rawurlencode( $post->ID ), $url );
 		$url = str_replace( '%post_url%', rawurlencode( $this->get_share_url( $post->ID ) ), $url );
 		$url = str_replace( '%post_full_url%', rawurlencode( get_permalink( $post->ID ) ), $url );
 		$url = str_replace( '%post_title%', rawurlencode( $this->get_share_title( $post->ID ) ), $url );
+		$url = str_replace( '%site_url%', rawurlencode( site_url() ), $url );
```

Comment:
I'd recommend using `home_url()` instead. `site_url()` returns the location of the WordPress files, not necessarily the site's home page.